plugins/modules/site_workflow_manager.py

```diff
@@ -797,13 +838,63 @@ def get_site_name(self, site):
                 self.set_operation_result("failed", False, self.msg, "ERROR")
                 return None
 
-            site_name = '/'.join([parent_name, name])
-            self.log("Constructed site name: {}".format(site_name), "INFO")
-            return site_name
+            site_name_hierarchy = '/'.join([parent_name, name])
+            self.log("Constructed site name: {}".format(site_name_hierarchy), "INFO")
+            return site_name_hierarchy
+
+        except Exception as e:
+            error_message = "An error occurred while getting site name: {}".format(str(e))
+            self.log(error_message, "ERROR")
+            return None
+
+    def get_bulk_site_names(self, site, bulk_operation=True):
+        """
+        Get and return a list of constructed site names for areas, buildings, and floors.
+
+        Parameters:
+        - self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+
+        Returns:
+        - list: A list of constructed site names.
+
+        Description:
+        This method collects and constructs site names from areas, buildings, and floors
+        based on the Cisco DNAC version.
+        """
+        name_list = []
+        try:
+            for area in self.config[0].get('site', {}).get('area', []):
+                area_name = area.get('name')
+                area_parent_name_hierarchy = area.get('parent_name_hierarchy')
+                if area_name and area_parent_name_hierarchy:
+                    name_list.append("{}/{}".format(area_parent_name_hierarchy, area_name))
+                elif not area_parent_name_hierarchy:
+                    self.log("Missing parent name hierarchy for area: {}".format(area_name), "ERROR")
+
+            for building in self.config[0].get('site', {}).get('building', []):
+                building_name = building.get('name')
+                building_parent_name_hierarchy = building.get('parent_name_hierarchy')
+                if building_name and building_parent_name_hierarchy:
+                    name_list.append("{}/{}".format(building_parent_name_hierarchy, building_name))
+                elif not building_parent_name_hierarchy:
+                    self.log("Missing parent name hierarchy for building: {}".format(building_name), "ERROR")
+
+            for floor in self.config[0].get('site', {}).get('floor', []):
+                floor_name = floor.get('name')
+                floor_parent_name_hierarchy = floor.get('parent_name_hierarchy')
+                if floor_name and floor_parent_name_hierarchy:
+                    name_list.append("{}/{}".format(floor_parent_name_hierarchy, floor_name))
+                elif not floor_parent_name_hierarchy:
+                    self.log("Missing parent name hierarchy for floor: {}".format(floor_name), "ERROR")
+
+            if not name_list:
```

Comment:
print the name_list as well.

        if not name_list:
            self.log("No site names constructed from areas, buildings, or floors.", "WARNING")
        else:
            self.log("Final constructed site names: {}".format(name_list), "DEBUG")
            self.log("Bulk site names construction completed successfully.", "DEBUG")

        return name_list

Reply:
Added these lines